Content.Server/Administration/Systems/AdminVerbSystem.cs

```diff
@@ -348,7 +348,30 @@ private void AddAdminVerbs(GetVerbsEvent<Verb> args)
                     Impact = LogImpact.Low
                 });
 
-                if (TryComp<SiliconLawBoundComponent>(args.Target, out var lawBoundComponent))
+                // This logic is needed to be able to modify the AI's laws through its core and eye.
+                SiliconLawBoundComponent? lawBoundComponent = null;
+
+                var target = args.Target;
+                TryComp(args.Target, out lawBoundComponent);
+
+                if (lawBoundComponent == null)
+                {
+                    // When inspecting the core we can find the entity with its laws by looking at the  AiHolderComponent.
+                    if (TryComp<StationAiHolderComponent>(args.Target, out var holder) && holder.Slot.Item != null
+                        && TryComp(holder.Slot.Item, out lawBoundComponent))
+                    {
+                        target = holder.Slot.Item.Value;
+                    }
+                    // For the eye we can find the entity with its laws as the source of the movement relay since the eye
+                    // is just a proxy for it to move around and look around the station.
+                    else if (TryComp<MovementRelayTargetComponent>(args.Target, out var relay)
+                        && TryComp(relay.Source, out lawBoundComponent))
+                    {
+                        target = relay.Source;
+                    }
+                }
+
+                if (lawBoundComponent != null)
```

Comment:
Why isn't this just an if-else with trycomp?

Reply:
Several iterations of the code just kind a led to it /shrug.